src/main/webapp/js/datepicker.js

```diff
@@ -89,7 +93,8 @@ function getMaxDateForVisibleDate(startDate, publishDate) {
 
 /**
  * @assumption: visibleDate has a valid value
- * @returns {Date} visibleDate
+ * @param {Date} visibleDate
+ * @returns {Date}
  */
 function getMinDateForPublishDate(visibleDate) {
     return visibleDate;
```

Comment:
> Here the function is used to set minDate option of ('#publishDate').datepicker() but there is no minDate option.

Take a look at Jquery datepicker https://jqueryui.com/datepicker/

> Also why publishDate has relation with visibleDate?
`publishDate` does not relate to `visibleDate`, only `visibledate` will affect the selection of `publishDate`
I guess this is you means: `visibledate` related to `startdate` and `publishdate` (as you don't want to `visibledate` after `startdate` or `publishdate`)